src/main/java/io/strimzi/kafka/metrics/DataPointSnapshotBuilder.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.metrics;
+
+import com.yammer.metrics.core.Histogram;
+import com.yammer.metrics.core.Sampling;
+import com.yammer.metrics.core.Timer;
+import io.prometheus.metrics.model.snapshots.CounterSnapshot;
+import io.prometheus.metrics.model.snapshots.GaugeSnapshot;
+import io.prometheus.metrics.model.snapshots.InfoSnapshot;
+import io.prometheus.metrics.model.snapshots.Labels;
+import io.prometheus.metrics.model.snapshots.PrometheusNaming;
+import io.prometheus.metrics.model.snapshots.Quantile;
+import io.prometheus.metrics.model.snapshots.Quantiles;
+import io.prometheus.metrics.model.snapshots.SummarySnapshot;
+
+import java.util.Arrays;
+
+/**
+ * Builder methods to convert Kafka metrics into Prometheus metrics
+ */
+public class DataPointSnapshotBuilder {
+
+    /**
+     * Convert a non numeric metric into a datapoint for a {@link InfoSnapshot} metric
+     * @param value The value of the datapoint
+     * @param labels Labels associated with the datapoint
+     * @param metricName The name of the new label
+     * @return The {@link InfoSnapshot.InfoDataPointSnapshot} datapoint
+     */
+    public static InfoSnapshot.InfoDataPointSnapshot convert(Object value, Labels labels, String metricName) {
+        labels = labels.add(PrometheusNaming.sanitizeLabelName(metricName), String.valueOf(value));
```

Comment:
do we want to mutate the caller's labels?

Reply:
`Labels.add()` returns a new instance, but yeah we don't want to reassign `labels`. Fixed.